### PR TITLE
Build the java query DSL api docs from a test

### LINKED ISSUE #24320: QueryDSLDocumentationTests should be imported into the java API docs
Right now there is a big comment at the top of QueryDSLDocumentationTests saying that we hope that they lines in there match the docs in the java API documentation. We should really use the `include-tagged::{doc-tests}/DeleteDocumentationIT.java[delete-request]` style syntax to include it directly into the asciidoc. The tags that you'd end up putting in `QueryDSLDocumentationTests` would make it *super* clear that this file is included in documentation. That way someone who missed the comment won't break the tests.

### PR DESCRIPTION
**Pull request by nik9000**
We've had `QueryDSLDocumentationTests` for a while but it had a very
hopeful comment at the top about how we want to make sure that the
example in the query-dsl docs match up with the test but we never
had anything that made *sure* that they did. This changes that!

Now the examples from the query-dsl docs are all built from the
`QueryDSLDocumentationTests`. All except for the percolator example
because that is hard to do as it stands now.

To make this easier this change moves `QueryDSLDocumentationTests`
from core and into the high level rest client. This is useful for
two reasons:
1. We expect the high level rest client to be able to use the builders.
2. The code that builds that docs doesn't check out all of
Elasticsearch. It only checks out certain directories. Since we're
already including snippets from that directory we don't have to
make any changes to that process.

Closes #24320

### CODE REVIEW

File: client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/QueryDSLDocumentationTests.java [hunk 1]
```diff
@@ -0,0 +1,453 @@
...
+
+/**
+ * Examples of using the transport client that are imported into the transport client documentation.
+ * There are no assertions here because we're mostly concerned with making sure that the examples
+ * compile and don't throw weird runtime exceptions. Assertions and example data would be nice, but
+ * that is secondary.
+ */
+public class QueryDSLDocumentationTests extends ESTestCase {
+    public void testBool() {
+        // tag::bool
```
Comment (nik9000) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/QueryDSLDocumentationTests.java]: Grumble.... I moved this file to make it easier for the docs framework to pick up but that made this much harder to review.....

**Review by clintongormley — APPROVED** [review]
Great work!

**Review by dadoonet — APPROVED** [review]
That is AWESOME!
Thank you so much Nik for doing it!

**Comment by nik9000** [comment]
Thanks for the reviews! I need to get elastic/docs#172 in or this will break the docs build but I think that is almost ready.

**Comment by nik9000** [comment]
I'm not backporting this to 5.x because we don't have the high-level-rest client in 5.x and that is where the test lives.

**Pull request merged**